manifests/init.pp

```diff
@@ -1,64 +1,64 @@
-  $package_provider                                       = $dhcp::params::package_provider,
-  Integer[0,65535] $ldap_port                             = 389,
-  String $ldap_server                                     = 'localhost',
-  String $ldap_username                                   = 'cn=root, dc=example, dc=com',
-  String $ldap_password                                   = '',
-  String $ldap_base_dn                                    = 'dc=example, dc=com',
-  Enum['dynamic', 'static'] $ldap_method                  = 'dynamic',
-  Optional[Stdlib::Absolutepath] $ldap_debug_file         = undef,
-  Boolean $use_ldap                                       = false,
-  String $option_code150_label                            = 'pxegrub',
-  String $option_code150_value                            = 'text',
-  Hash[String, Hash] $dhcp_classes                        = {},
-  Hash[String, Hash] $hosts                               = {},
-  Hash[String, Hash] $ignoredsubnets                      = {},
-  Hash[String, Hash] $pools                               = {},
-  Hash[String, Hash] $pools6                              = {},
-  Optional[Stdlib::Absolutepath] $dhcpd_binary            = $dhcp::params::dhcpd_binary
+  Optional[Array[String[1]]] $dnsdomain                            = undef,
+  Array[Stdlib::IP::Address::V4] $nameservers                      = [],
+  Array[Stdlib::IP::Address::V6] $nameservers_ipv6                 = [],
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]] $ntpservers     = [],
+  Array[Stdlib::Fqdn] $dnssearchdomains                            = [],
+  String[1] $dhcp_conf_header                                      = 'INTERNAL_TEMPLATE',
+  String[1] $dhcp_conf_ddns                                        = 'INTERNAL_TEMPLATE',
+  String[1] $dhcp_conf_ntp                                         = 'INTERNAL_TEMPLATE',
+  String[1] $dhcp_conf_pxe                                         = 'INTERNAL_TEMPLATE',
+  String[1] $dhcp_conf_extra                                       = 'INTERNAL_TEMPLATE',
+  Hash[String[1], Hash[String[1], String[1]]] $dhcp_conf_fragments = {},
+  Optional[Array[String[1]]] $interfaces                           = undef,
+  String[1] $interface                                             = 'NOTSET',
+  Optional[String[1]] $dnsupdatekey                                = undef,
+  String[1] $ddns_update_style                                     = 'interim',
+  Optional[String[1]] $dnskeyname                                  = undef,
+  String[1] $ddns_update_static                                    = 'on',
+  String[1] $ddns_update_optimize                                  = 'on',
+  Enum['allow', 'deny'] $ddns_client_updates                       = 'allow',
+  Optional[Stdlib::IP::Address] $pxeserver                         = undef,
+  Optional[String[1]] $pxefilename                                 = undef,
+  Optional[Integer[1]] $mtu                                        = undef,
```

Comment:
An MTU of `1` is no more valid than `0`

Reply:
I'm not sure which is the lowest positive number that makes sense, I just wanted to prohibit negative numbers.